### PR TITLE
XRDDEV-1961 - disabled messagelog API and UI changes

### PR DESCRIPTION
**Pull request by amoog**
Contains  both 1962 and 1963

### CODE REVIEW

**Review by jansu76 — CHANGES_REQUESTED** [review]

File: src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java [hunk 1]
```diff
@@ -113,6 +114,8 @@ public final class ProxyMain {
 
     private static final int STATS_LOG_REPEAT_INTERVAL = 60;
 
+    private static AddOnStatusDiagnostics addOnStatus;
```
Comment (ricardas-buc) [inline src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java:117]: @jansu76 it seems that this is almost god class. Were there any discussions about making it more granular?

Comment (jansu76) [reply src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java:117]: I am not aware of any discussions. Refactoring might do it good, e.g. moving the different handlers to separate classes. As far as god classes go though, I'm afraid this isn't even close to the level of worst ones...

The changes from this PR are quite small so I think I would outscope refactoring to a separate ticket (or if you are real keen on doing those now, I can let you do the modifications).

Comment (ricardas-buc) [reply src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java:117]: Understood, for sure it's out of scope, but was curious.

File: src/proxy-ui-api/src/main/resources/openapi-definition.yaml [hunk 2]
```diff
@@ -2095,6 +2095,32 @@ paths:
           description: request specified an invalid format
         '500':
           description: internal server error
+  /diagnostics/addon-status:
```
[the remark this answers is not in the document]
Comment (jansu76) [reply src/proxy-ui-api/src/main/resources/openapi-definition.yaml:2098]: Addressed in https://github.com/nordic-institute/X-Road/commit/6a509c23060d879ea0b58f9b970b6ebf2b7f4294

File: src/proxy-ui-api/src/main/resources/openapi-definition.yaml [hunk 3]
```diff
@@ -5219,6 +5245,14 @@ components:
...
           items:
             $ref: '#/components/schemas/AccessRight'
+    AddonStatus:
+      type: object
+      description: addon module statuses
+      required:
+        - messagelog_enabled
+      properties:
+        messagelog_enabled:
+          type: boolean
```
[the remark this answers is not in the document]
Comment (jansu76) [reply src/proxy-ui-api/src/main/resources/openapi-definition.yaml:5255]: Addressed in https://github.com/nordic-institute/X-Road/commit/6a509c23060d879ea0b58f9b970b6ebf2b7f4294

File: src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java [hunk 4]
```diff
@@ -233,9 +238,29 @@ private static AdminPort createAdminPort() throws Exception {
...
+
         return adminPort;
     }
 
+    private static void logResponseIOError(IOException e) {
+        log.error("Unable to write to provided response, delegated request handling failed, response may"
+                + " be malformed", e);
+    }
+
+    private static void addAddonStatusHandler(AdminPort adminPort) {
```
[the remark this answers is not in the document]
Comment (jansu76) [reply src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java]: Addressed in https://github.com/nordic-institute/X-Road/commit/6a509c23060d879ea0b58f9b970b6ebf2b7f4294

File: src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java [hunk 5]
```diff
@@ -233,9 +238,29 @@ private static AdminPort createAdminPort() throws Exception {
...
+        adminPort.addHandler("/addonstatus", new AdminPort.SynchronousCallback() {
+            @Override
+            public void handle(HttpServletRequest request, HttpServletResponse response) {
+                try {
+                    JsonUtils.getSerializer().toJson(addOnStatus, response.getWriter());
```
[the remark this answers is not in the document]
Comment (jansu76) [reply src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java:257]: Addressed in https://github.com/nordic-institute/X-Road/commit/6a509c23060d879ea0b58f9b970b6ebf2b7f4294

File: src/proxy-ui-api/src/main/java/org/niis/xroad/securityserver/restapi/openapi/DiagnosticsApiController.java [hunk 6]
```diff
@@ -82,4 +86,11 @@ public ResponseEntity<Set<OcspResponderDiagnostics>> getOcspRespondersDiagnostic
         List<OcspResponderDiagnosticsStatus> statuses = diagnosticService.queryOcspResponderStatus();
         return new ResponseEntity<>(ocspResponderDiagnosticConverter.convert(statuses), HttpStatus.OK);
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('DIAGNOSTICS')")
+    public ResponseEntity<AddonStatus> getAddonDiagnostics() {
```
[the remark this answers is not in the document]
Comment (jansu76) [reply src/proxy-ui-api/src/main/java/org/niis/xroad/securityserver/restapi/openapi/DiagnosticsApiController.java]: Addressed in https://github.com/nordic-institute/X-Road/commit/6a509c23060d879ea0b58f9b970b6ebf2b7f4294

**Review by ricardas-buc — APPROVED** [review]

**Review by jansu76 — APPROVED** [review]

**Pull request merged**